diff --git a/DaoGenerator/src-template/dao-master.ftl b/DaoGenerator/src-template/dao-master.ftl
index adaffc4..d44e883 100644
--- a/DaoGenerator/src-template/dao-master.ftl
+++ b/DaoGenerator/src-template/dao-master.ftl
@@ -26,6 +26,7 @@ import android.database.sqlite.SQLiteOpenHelper;
 import android.util.Log;
 import de.greenrobot.dao.AbstractDaoMaster;
 import de.greenrobot.dao.identityscope.IdentityScopeType;
+<#if schema.extraDaoImport??>import ${schema.extraDaoImport};</#if>
 
 <#list schema.entities as entity>
 import ${entity.javaPackageDao}.${entity.classNameDao};
@@ -58,7 +59,10 @@ public class DaoMaster extends AbstractDaoMaster {
     
     public static abstract class OpenHelper extends SQLiteOpenHelper {
 
+	private Context con;
+
         public OpenHelper(Context context, String name, CursorFactory factory) {
+	    this.con = context;
             super(context, name, factory, SCHEMA_VERSION);
         }
 
@@ -66,6 +70,7 @@ public class DaoMaster extends AbstractDaoMaster {
         public void onCreate(SQLiteDatabase db) {
             Log.i("greenDAO", "Creating tables for schema version " + SCHEMA_VERSION);
             createAllTables(db, false);
+	    ExtraDaoMater.onCreate(db, con);
         }
     }
     
diff --git a/DaoGenerator/src/de/greenrobot/daogenerator/Schema.java b/DaoGenerator/src/de/greenrobot/daogenerator/Schema.java
index 8351ecb..366ee14 100644
--- a/DaoGenerator/src/de/greenrobot/daogenerator/Schema.java
+++ b/DaoGenerator/src/de/greenrobot/daogenerator/Schema.java
@@ -39,6 +39,8 @@ public class Schema {
     private boolean hasKeepSectionsByDefault;
     private boolean useActiveEntitiesByDefault;
 
+    private String extraDaoImport;
+
     public Schema(int version, String defaultJavaPackage) {
         this.version = version;
         this.defaultJavaPackage = defaultJavaPackage;
@@ -46,6 +48,14 @@ public class Schema {
         initTypeMappings();
     }
 
+    public String getExtraDaoImport() {
+	return extraDaoImport;
+    }
+
+    public void setExtraDaoImport(String extraDaoImport) {
+	this.extraDaoImport = extraDaoImport;
+    }
+
     public void enableKeepSectionsByDefault() {
         hasKeepSectionsByDefault = true;
     }
